### ISSUE TITLE
Limit after offset resets it

### ISSUE THREAD

**Issue by smaant** [funcool/suricatta#9]
```clojure
> (-> (dsl/select :name)
      (dsl/from :book)
      (dsl/offset 100)
      (dsl/limit 10)
      (fmt/get-sql))

"select name from book limit ?"
```

This is because of:
> SelectOffsetStep<R> limit(int numberOfRows)
> This is the same as calling limit(int, int) with offset = 0, or calling .limit(numberOfRows).offset(0)

http://www.jooq.org/javadoc/latest/org/jooq/SelectLimitStep.html#limit-int-

**Comment by niwinz** [comment]
I'm not clearly understand the issue. the limit should be assigned before offset as in sql, so if you are doing it in different order, the behavior is not defined.

**Comment by smaant** [comment]
Nope, both sql queries
```sql
select * from table offset 10 limit 5;
select * from table limit 5 offset 10;
```
brings the same result which is not the case for dsl and it's very unexpected, why would `limit` change `offset`.

**Comment by niwinz** [comment]
Oh, you are right! It seems a jOOQ bug. Thanks for report it.

**Comment by smaant** [comment]
Sure, no problem. Are you going to report this to jOOQ or will make a workaround in suricatta?

**Comment by niwinz** [comment]
I'm just writing a mail to the jOOQ mailing list. It is not easy workaround from suricatta part :(

**Comment by smaant** [comment]
Yea, I understand. Anyway, it's not a serious issue and there's an obvious workaround I can do in my code.

**Comment by lukaseder** [comment]
Thanks for your reports and discussions. The API usage that produced this bug is not possible in plain Java, but it certainly is in Clojure, which can work with Java APIs in a more relaxed, less type safe way.

I've added a fix due for jOOQ 3.8, which allows inversing `OFFSET .. LIMIT` as it is allowed in some databases: https://github.com/jOOQ/jOOQ/issues/4784.

In the meantime, I suspect that restricting use to `LIMIT .. OFFSET` (in that order) is the only solution right now.

**Comment by niwinz** [comment]
Nice! thanks for the quick reply @lukaseder !

**Comment by niwinz** [comment]
Should be fixed in 1.0.0 as it uses jOOQ 3.8.1
